### PR TITLE
test(lint): refactor logs and add linting rules

### PR DESCRIPTION
**Pull request by laurentiuNiculae**
**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by rchincha** [comment]
Think about this pattern ...

pkg/log/constants.go

const RepoNotFoundMsg = "repo not found"

log.Info().Msg(RepoNotFoundMsg)

**Comment by rchincha** [comment]
Error msgs ... to be standardized.

"Failed to ..."
"Error happened ..."
"Unable to ..."

**Comment by rchincha** [comment]
Add a log linter if possible.

File: LINT.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
```
Comment (laurentiuNiculae) [inline LINT.sh:1]: > **Reminder**
> Delete this file after you create the check-logs make target

File: errors/errors.go [hunk 2]
```diff
@@ -46,128 +46,126 @@ func GetDetails(err error) map[string]string {
...
-	ErrTagMetaNotFound                = errors.New("metadb: tag metadata not found for given repo and tag names")
-	ErrTypeAssertionFailed            = errors.New("storage: failed DatabaseDriver type assertion")
-	ErrInvalidRequestParams           = errors.New("resolver: parameter sent has invalid value")
+	ErrBadConfig                      = errors.New("invalid server config")
+	ErrCliBadConfig                   = errors.New("invalid cli config")
+	ErrRepoNotFound                   = errors.New("repository not found")
+	ErrRepoBadVersion                 = errors.New("repository unsupported layout version")
+	ErrRepoBadLayout                  = errors.New("repository invalid layout")
+	ErrManifestNotFound               = errors.New("manifest not found")
+	ErrBadManifest                    = errors.New("manifest invalid contents")
```
Comment (andaaron) [inline errors/errors.go:55]:
```suggestion
	ErrBadManifest                    = errors.New("invalid manifest content")
```

File: errors/errors.go [hunk 3]
```diff
@@ -46,128 +46,126 @@ func GetDetails(err error) map[string]string {
...
-	ErrUnexpectedMediaType            = errors.New("metadb: got unexpected media type")
-	ErrRepoMetaNotFound               = errors.New("metadb: repo metadata not found for given repo name")
-	ErrTagMetaNotFound                = errors.New("metadb: tag metadata not found for given repo and tag names")
-	ErrTypeAssertionFailed            = errors.New("storage: failed DatabaseDriver type assertion")
-	ErrInvalidRequestParams           = errors.New("resolver: parameter sent has invalid value")
+	ErrBadConfig                      = errors.New("invalid server config")
+	ErrCliBadConfig                   = errors.New("invalid cli config")
+	ErrRepoNotFound                   = errors.New("repository not found")
+	ErrRepoBadVersion                 = errors.New("repository unsupported layout version")
+	ErrRepoBadLayout                  = errors.New("repository invalid layout")
```
Comment (andaaron) [inline errors/errors.go:53]:
```suggestion
	ErrRepoBadLayout                  = errors.New("invalid repository layout")
```

File: errors/errors.go [hunk 4]
```diff
@@ -46,128 +46,126 @@ func GetDetails(err error) map[string]string {
...
-	ErrImageMetaNotFound              = errors.New("metadb: image meta not found")
-	ErrUnexpectedMediaType            = errors.New("metadb: got unexpected media type")
-	ErrRepoMetaNotFound               = errors.New("metadb: repo metadata not found for given repo name")
-	ErrTagMetaNotFound                = errors.New("metadb: tag metadata not found for given repo and tag names")
-	ErrTypeAssertionFailed            = errors.New("storage: failed DatabaseDriver type assertion")
-	ErrInvalidRequestParams           = errors.New("resolver: parameter sent has invalid value")
+	ErrBadConfig                      = errors.New("invalid server config")
+	ErrCliBadConfig                   = errors.New("invalid cli config")
+	ErrRepoNotFound                   = errors.New("repository not found")
+	ErrRepoBadVersion                 = errors.New("repository unsupported layout version")
```
Comment (andaaron) [inline errors/errors.go:52]:
```suggestion
	ErrRepoBadVersion                 = errors.New("unsupported repository layout version")
```

File: errors/errors.go [hunk 5]
```diff
@@ -46,128 +46,126 @@ func GetDetails(err error) map[string]string {
...
+	ErrRepoBadLayout                  = errors.New("repository invalid layout")
+	ErrManifestNotFound               = errors.New("manifest not found")
+	ErrBadManifest                    = errors.New("manifest invalid contents")
+	ErrUploadNotFound                 = errors.New("uploads not found")
+	ErrBadUploadRange                 = errors.New("uploads bad range")
```
Comment (andaaron) [inline errors/errors.go:57]:
```suggestion
	ErrBadUploadRange                 = errors.New("bad upload Content-Length")
```

File: errors/errors.go [hunk 6]
```diff
@@ -46,128 +46,126 @@ func GetDetails(err error) map[string]string {
...
-	ErrTypeAssertionFailed            = errors.New("storage: failed DatabaseDriver type assertion")
-	ErrInvalidRequestParams           = errors.New("resolver: parameter sent has invalid value")
+	ErrBadConfig                      = errors.New("invalid server config")
+	ErrCliBadConfig                   = errors.New("invalid cli config")
+	ErrRepoNotFound                   = errors.New("repository not found")
+	ErrRepoBadVersion                 = errors.New("repository unsupported layout version")
+	ErrRepoBadLayout                  = errors.New("repository invalid layout")
+	ErrManifestNotFound               = errors.New("manifest not found")
+	ErrBadManifest                    = errors.New("manifest invalid contents")
+	ErrUploadNotFound                 = errors.New("uploads not found")
```
Comment (andaaron) [inline errors/errors.go:56]:
```suggestion
	ErrUploadNotFound                 = errors.New("upload destination not found")
```

File: errors/errors.go [hunk 7]
```diff
@@ -46,128 +46,126 @@ func GetDetails(err error) map[string]string {
...
+	ErrEntriesExceeded                = errors.New("too many entries returned")
+	ErrLDAPEmptyPassphrase            = errors.New("empty ldap passphrase")
+	ErrLDAPBadConn                    = errors.New("bad ldap connection")
+	ErrLDAPConfig                     = errors.New("invalid LDAP configuration")
+	ErrCacheRootBucket                = errors.New("unable to create/update root cache bucket")
+	ErrCacheNoBucket                  = errors.New("unable to find cache bucket")
+	ErrCacheMiss                      = errors.New("cache miss")
+	ErrRequireCred                    = errors.New("bind ldap credentials required")
+	ErrInvalidCred                    = errors.New("invalid ldap credentials")
+	ErrEmptyJSON                      = errors.New("cli config json is empty")
```
Comment (andaaron) [inline errors/errors.go:75]: How is this different from ErrCliBadConfig? Why not use that?

File: errors/errors.go [hunk 8]
```diff
@@ -46,128 +46,126 @@ func GetDetails(err error) map[string]string {
...
+	ErrInvalidArgs                    = errors.New("invalid cli arguments")
+	ErrInvalidFlagsCombination        = errors.New("invalid cli combination of flags")
+	ErrInvalidURL                     = errors.New("invalid URL format")
+	ErrExtensionNotEnabled            = errors.New("functionality is not built/configured in the current server")
+	ErrUnauthorizedAccess             = errors.New("unauthorized access. check credentials")
+	ErrCannotResetConfigKey           = errors.New("cannot reset given config key")
+	ErrConfigNotFound                 = errors.New("config with the given name does not exist")
+	ErrNoURLProvided                  = errors.New("no URL provided")
+	ErrIllegalConfigKey               = errors.New("given config key is not allowed")
+	ErrScanNotSupported               = errors.New("scanning of image media type not supported")
```
Comment (andaaron) [inline errors/errors.go:85]:
```suggestion
	ErrScanNotSupported               = errors.New("scanning is not supported for given image media type")
```

File: errors/errors.go [hunk 9]
```diff
@@ -46,128 +46,126 @@ func GetDetails(err error) map[string]string {
...
+	ErrEmptyTag                       = errors.New("tag can't be empty string")
+	ErrEmptyDigest                    = errors.New("digest can't be empty string")
 	ErrInvalidRepoRefFormat           = errors.New("invalid image reference format [repo:tag] or [repo@digest]")
-	ErrLimitIsNegative                = errors.New("pagination: limit has negative value")
-	ErrOffsetIsNegative               = errors.New("pagination: offset has negative value")
-	ErrSortCriteriaNotSupported       = errors.New("pagination: the sort criteria is not supported")
-	ErrMediaTypeNotSupported          = errors.New("metadb: media type is not supported")
+	ErrLimitIsNegative                = errors.New("pagination limit has negative value")
+	ErrOffsetIsNegative               = errors.New("pagination offset has negative value")
+	ErrSortCriteriaNotSupported       = errors.New("pagination the sort criteria is not supported")
```
Comment (andaaron) [inline errors/errors.go:122]:
```suggestion
	ErrSortCriteriaNotSupported       = errors.New("the pagination sort criteria is not supported")
```

File: errors/errors.go [hunk 10]
```diff
@@ -46,128 +46,126 @@ func GetDetails(err error) map[string]string {
...
-	ErrBucketDoesNotExist             = errors.New("DB: bucket does not exist")
-	ErrOpenIDProviderDoesNotExist     = errors.New("openID: provider does not exist in given config")
-	ErrHashKeyNotCreated              = errors.New("cookiestore: generated random hash key is nil, not continuing")
+	ErrDedupeRebuild                  = errors.New("couldn't rebuild dedupe index")
+	ErrMissingAuthHeader              = errors.New("required authorization header is missing")
+	ErrUserAPIKeyNotFound             = errors.New("user info for given API key hash not found")
+	ErrUserSessionNotFound            = errors.New("user session for given ID not found")
+	ErrBucketDoesNotExist             = errors.New("bucket does not exist")
+	ErrOpenIDProviderDoesNotExist     = errors.New("openid provider does not exist in given config")
+	ErrHashKeyNotCreated              = errors.New("cookiestore generated random hash key is nil, not continuing")
```
Comment (andaaron) [inline errors/errors.go:132]:
```suggestion
	ErrHashKeyNotCreated              = errors.New("cookiestore generated random hash key is nil, aborting")
```

File: errors/errors.go [hunk 11]
```diff
@@ -46,128 +46,126 @@ func GetDetails(err error) map[string]string {
...
+	ErrCouldNotMarshalBookmarkedRepos = errors.New("could not repack entry for user bookmarked repos")
+	ErrUserDataNotFound               = errors.New("user data not found for given user identifier")
+	ErrUserDataNotAllowed             = errors.New("user data operations are not allowed")
+	ErrCouldNotPersistData            = errors.New("could not persist to db")
+	ErrSignConfigDirNotSet            = errors.New("signature config dir not set")
+	ErrBadSignatureManifestDigest     = errors.New("bad signature manifest digest")
+	ErrInvalidSignatureType           = errors.New("invalid signature type")
+	ErrSyncPingRegistry               = errors.New("unable to ping any registry URLs")
+	ErrSyncImageNotSigned             = errors.New("synced image is not signed")
+	ErrSyncImageFilteredOut           = errors.New("sync image is filtered out by sync config")
```
Comment (andaaron) [inline errors/errors.go:147]:
```suggestion
	ErrSyncImageFilteredOut           = errors.New("image is filtered out by sync config")
```

File: errors/errors.go (hunk 7 above)
Comment (laurentiuNiculae) [reply errors/errors.go:75]: I think it's more specific, maybe we should return a merge of ErrCliBadConfig with ErrEmptyJSON

File: pkg/meta/boltdb/boltdb.go
Comment (andaaron) [inline pkg/meta/boltdb/boltdb.go:1]: There's a lot of custom errors in this file. Can we wrap errors defined in errors.go together with errors coming from boltdb instead?

File: pkg/meta/dynamodb/dynamodb.go
Comment (andaaron) [inline pkg/meta/dynamodb/dynamodb.go:1]: There's a lot of custom errors in this file. Can we wrap errors defined in errors.go together with errors coming from dynamodb instead?

File: pkg/meta/hooks.go [hunk 12]
```diff
@@ -70,27 +70,29 @@ func OnDeleteManifest(repo, reference, mediaType string, digest godigest.Digest,
...
 		if err != nil {
-			log.Info().Msg("metadb: restoring image store")
+			log.Info().Str("component", "metadb").Msg("restoring image store")
 
 			// restore image store
 			_, _, err := imgStore.PutImageManifest(repo, reference, mediaType, manifestBlob)
 			if err != nil {
-				log.Error().Err(err).Msg("metadb: error while restoring image store, database is not consistent")
+				log.Error().Err(err).Str("component", "metadb").
+					Msg("error while restoring image store, database is not consistent")
```
Comment (andaaron) [inline pkg/meta/hooks.go:86]:
```suggestion
					Msg("failed to restore manifest to image store, database is not consistent")
```

File: pkg/meta/parse.go [hunk 13]
```diff
@@ -314,7 +316,7 @@ func SetImageMetaFromInput(ctx context.Context, repo, reference, mediaType strin
 
 		err := json.Unmarshal(blob, &manifestContent)
 		if err != nil {
-			log.Error().Err(err).Msg("metadb: error while getting image data")
+			log.Error().Err(err).Str("component", "metadb").Msg("error while getting image data")
```
Comment (andaaron) [inline pkg/meta/parse.go:319]:
```suggestion
			log.Error().Err(err).Str("component", "metadb").Msg("failed to unmarshal image manifest")
```

File: pkg/meta/parse.go [hunk 14]
```diff
@@ -348,15 +350,15 @@ func SetImageMetaFromInput(ctx context.Context, repo, reference, mediaType strin
...
+					Msg("failed set signature meta for signed image")
 
 				return err
 			}
 
 			err = metaDB.UpdateSignaturesValidity(ctx, repo, signedManifestDigest)
 			if err != nil {
 				log.Error().Err(err).Str("repository", repo).Str("reference", reference).Str("digest",
-					signedManifestDigest.String()).Msg("load-repo: failed verify signatures validity for signed image")
+					signedManifestDigest.String()).Msg("failed verify signatures validity for signed image")
```
Comment (andaaron) [inline pkg/meta/parse.go:361]:
```suggestion
					signedManifestDigest.String()).Msg("failed to verify signature validity for signed image")
```

File: pkg/meta/parse.go [hunk 15]
```diff
@@ -380,7 +382,7 @@ func SetImageMetaFromInput(ctx context.Context, repo, reference, mediaType strin
 
 	err := metaDB.SetRepoReference(ctx, repo, reference, imageMeta)
 	if err != nil {
-		log.Error().Err(err).Msg("metadb: error while setting repo meta")
+		log.Error().Err(err).Str("component", "metadb").Msg("error while setting repo meta")
```
Comment (andaaron) [inline pkg/meta/parse.go:385]:
```suggestion
		log.Error().Err(err).Str("component", "metadb").Msg("failed to set repo meta")
```

File: pkg/scheduler/scheduler.go
Comment (andaaron) [inline pkg/scheduler/scheduler.go:1]: Careful with this one, as Alexei has another PR which probably would conflict.

File: pkg/storage/common/common.go [hunk 16]
```diff
@@ -1086,7 +1087,7 @@ func (gen *StorageMetricsInitGenerator) Next() (scheduler.Task, error) {
 		return nil, err
 	}
 
-	gen.Log.Debug().Str("repo", repo).Int("randomDelay", delay).Msg("StorageMetricsInitGenerator")
+	gen.Log.Debug().Str("repo", repo).Int("randomDelay", delay).Msg("storageMetricsInitGenerator")
```
Comment (andaaron) [inline pkg/storage/common/common.go:1090]:
```suggestion
	gen.Log.Debug().Str("repo", repo).Int("randomDelay", delay).Msg("generate task for storage metrics")
```

File: scripts/check_logs.sh [hunk 17]
```diff
@@ -0,0 +1,78 @@
...
+            "$(echo $line | tr -s [:space:])"
+        done <<< "$lintOutput"
+    fi
+done
+
+if [ $found_linting_error = true ]; then
+    exit 1
+fi
+
+exit 0
```
Comment (andaaron) [inline scripts/check_logs.sh:78]: Add a new line at the end of the file

File: errors/errors.go (hunk 7 above)
Comment (andaaron) [reply errors/errors.go:75]: I thought we wanted the errors a bit more generic?

File: pkg/meta/boltdb/boltdb.go [hunk 18]
```diff
@@ -494,7 +494,7 @@ func (bdw *BoltDB) SearchTags(ctx context.Context, searchText string,
 	searchedRepo, searchedTag, err := common.GetRepoTag(searchText)
 	if err != nil {
 		return []mTypes.FullImageMeta{},
-			fmt.Errorf("metadb: error while parsing search text, invalid format %w", err)
+			fmt.Errorf("failled to parse search text, invalid format %w", err)
```
Comment (andaaron) [inline pkg/meta/boltdb/boltdb.go:497]:
```suggestion
			fmt.Errorf("failed to parse search text, invalid format %w", err)
```

File: pkg/meta/boltdb/boltdb.go [hunk 19]
```diff
@@ -538,7 +538,7 @@ func (bdw *BoltDB) SearchTags(ctx context.Context, searchText string,
 
 				imageManifestData, err := getProtoImageMeta(imageBuck, manifestDigest)
 				if err != nil {
-					return fmt.Errorf("metadb: error fetching manifest meta for manifest with digest %s %w",
+					return fmt.Errorf("error fetching manifest meta for manifest with digest %s %w",
```
Comment (andaaron) [inline pkg/meta/boltdb/boltdb.go:541]:
```suggestion
					return fmt.Errorf("failed to fetch manifest meta for manifest with digest %s %w",
```

File: pkg/meta/boltdb/boltdb.go [hunk 20]
```diff
@@ -548,7 +548,7 @@ func (bdw *BoltDB) SearchTags(ctx context.Context, searchText string,
 
 				imageIndexData, err := getProtoImageMeta(imageBuck, indexDigest)
 				if err != nil {
-					return fmt.Errorf("metadb: error fetching manifest meta for manifest with digest %s %w",
+					return fmt.Errorf("error fetching manifest meta for manifest with digest %s %w",
```
Comment (andaaron) [inline pkg/meta/boltdb/boltdb.go:551]:
```suggestion
					return fmt.Errorf("failed to fetch manifest meta for manifest with digest %s %w",
```

File: pkg/meta/dynamodb/dynamodb.go [hunk 21]
```diff
@@ -637,7 +637,7 @@ func (dwr *DynamoDB) SearchTags(ctx context.Context, searchText string) ([]mType
 			imageIndexData, err := dwr.GetProtoImageMeta(ctx, indexDigest)
 			if err != nil {
 				return []mTypes.FullImageMeta{},
-					fmt.Errorf("metadb: error fetching manifest meta for manifest with digest %s %w", indexDigest, err)
+					fmt.Errorf("error fetching manifest meta for manifest with digest %s %w", indexDigest, err)
```
Comment (andaaron) [inline pkg/meta/dynamodb/dynamodb.go:640]:
```suggestion
					fmt.Errorf("failed to fetch manifest meta for manifest with digest %s %w", indexDigest, err)
```

File: pkg/storage/common/common.go [hunk 22]
```diff
@@ -978,14 +978,14 @@ func (gen *DedupeTaskGenerator) Next() (scheduler.Task, error) {
 		gen.repos, err = gen.ImgStore.GetRepositories()
 		if err != nil {
 			//nolint: dupword
-			gen.Log.Error().Err(err).Msg("dedupe rebuild: unable to to get list of repositories")
+			gen.Log.Error().Err(err).Str("component", "dedupe").Msg("unable to to get list of repositories")
```
Comment (andaaron) [inline pkg/storage/common/common.go:981]:
```suggestion
			gen.Log.Error().Err(err).Str("component", "dedupe").Msg("failed to to get list of repositories")
```

**Review by andaaron — COMMENTED** [review]
Otherwise looks ok

File: pkg/storage/imagestore/imagestore.go [hunk 23]
```diff
@@ -144,14 +144,14 @@ func (is *ImageStore) initRepo(name string) error {
 	// create "blobs" subdir
 	err := is.storeDriver.EnsureDir(path.Join(repoDir, "blobs"))
 	if err != nil {
-		is.log.Error().Err(err).Msg("error creating blobs subdir")
+		is.log.Error().Err(err).Str("repository", repoDir).Msg("failed to create blobs subdir")
```
Comment (peusebiu) [inline pkg/storage/imagestore/imagestore.go:147]: ("repository", name)

**Pull request merged**